nncf/dynamic_graph/input_wrapping.py

```diff
@@ -0,0 +1,121 @@
+from collections import OrderedDict
+from inspect import Signature, Parameter
+from typing import List
+
+import torch
+
+from nncf.dynamic_graph.patch_pytorch import register_operator
+from nncf.dynamic_graph.graph_builder import ModelInputInfo, create_mock_tensor
+from nncf.utils import is_tensor, objwalk
+from nncf.nncf_logger import logger as nncf_logger
+
+
+@register_operator()
+def nncf_model_input(tensor: 'torch.Tensor'):
+    return tensor
+
+
+def wrap_nncf_model_inputs_helper(model_args, model_kwargs, input_infos: List['ModelInputInfo'] = None,
+                                  force_objwalk=False):
+    """Will wrap the input tensors among the model_args and model_kwargs with a no-op required for
+    correct input tracing by NNCF. If input_infos is supplied and force_objwalk is False, will only
+    affect the tensor args/kwargs as specified in the input_infos (usually populated via a corresponding
+    section in the NNCF config that defines input tensor shapes and types). Otherwise will recursively
+    search model_args and model_kwargs for tensors and apply the wrapping for each found tensor."""
+    if input_infos is not None and not force_objwalk:
+        input_info_args_indices = [i for i, info in enumerate(input_infos) if info.keyword is None]
+        input_info_kwargs = [info.keyword for info in input_infos if info.keyword is not None]
+        args_list = list(model_args)
+        for idx, _ in enumerate(input_info_args_indices):
+            if idx < len(args_list):
+                arg = args_list[idx]
+                if is_tensor(arg):
+                    args_list[idx] = nncf_model_input(arg)
+        for kw in input_info_kwargs:
+            if is_tensor(model_kwargs[kw]):
+                model_kwargs[kw] = nncf_model_input(model_kwargs[kw])
+
+        model_args = tuple(args_list)
+    else:
+        # Fallback - ideally the input_infos must be supplied
+        model_args = objwalk(model_args, is_tensor, nncf_model_input)
+        model_kwargs = objwalk(model_kwargs, is_tensor, nncf_model_input)
+    return model_args, model_kwargs
+
+
+def wrap_nncf_model_inputs_with_objwalk(model_args, model_kwargs):
```

Comment:
Is this used anywhere? Or expected to be used?

Reply:
May be useful in the future, IMO.

Reply:
BTW, it can be called in the `else` branch of `wrap_nncf_model_inputs_helper`

Reply:
Refactored to use this in `default_dummy_forward` if no `wrap_inputs_fn` is supplied.